### PR TITLE
Use nginx-ingress for k3s

### PR DESCRIPTION
**Pull request by HotThoughts**
- [x] setup k3s with ingress-nginx
- [x] install Rasa X with enabled ingress using helmfile
- [x] run tests that use the ingress resource

### CODE REVIEW

File: .github/deployments/nginx-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: extensions/v1beta1
```
Comment (tczekajlo) [inline .github/deployments/nginx-ingress.yaml:1]: Why do we need it?

Comment (HotThoughts) [reply .github/deployments/nginx-ingress.yaml:1]: Removed 👍

File: .github/deployments/helmfile.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          enabled: false
+      - ingress:
+          annotations:
+            kubernetes.io/ingress.class: nginx
+            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
+            nginx.ingress.kubernetes.io/enable-cors: "true"
+          hosts:
+            - host: example.test.com
+              paths:
+                - /
```
Comment (tczekajlo) [inline .github/deployments/helmfile.yaml:21]: `No newline at and of file`

Comment (tczekajlo) [inline .github/deployments/helmfile.yaml:21]:
```suggestion
      - rasax:
          disableTelemetry: true
          initialUser:
            password: {{ requiredEnv "INITIAL_USER_PASSWORD" }}
      - duckling:
          enabled: false
      - nginx:
          enabled: false
      - ingress:
          annotations:
            kubernetes.io/ingress.class: nginx
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/enable-cors: "true"
          hosts:
            - host: example.test.com
              paths:
                - /
```
You need a few options more

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -88,40 +92,46 @@ jobs:
...
-          --set ingress.hosts[0].host=,ingress.hosts[0].paths={/} \
-          ${RELEASE_NAME} ${CHART}
-
-    - name: Wait for deployment to be ready
+        helmfile -f .github/deployments/helmfile.yaml sync
+
+    - name: Set host name and address env variables 🖌
+      run: |
+        ADDRESS=$(kubectl get ingress  -o json | jq .items[0].status.loadBalancer.ingress[0].ip)
+        HOST=$(kubectl get ingress  -o json | jq .items[0].spec.rules[0].host)
```
Comment (tczekajlo) [inline .github/workflows/ci.yml:102]:
```suggestion
        ADDRESS=$(kubectl get ingress -o jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}')
        HOST=$(kubectl get ingress -o jsonpath='{.items[0].spec.rules[0].host}')
```

You don't have to use `jq`, `kubectl` can be used to get a value of a specific path

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/deployments/rasa-x-helmfile.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          initialUser:
+            password: {{ requiredEnv "INITIAL_USER_PASSWORD" }}
+      - duckling:
+          enabled: false
+      - nginx:
+          enabled: false
+      - ingress:
+          annotations:
+            kubernetes.io/ingress.class: nginx
+            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
```
Comment (tczekajlo) [inline .github/deployments/rasa-x-helmfile.yaml:22]:
```suggestion
```

We don't use HTTPS for tests

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -88,40 +92,69 @@ jobs:
         docker rmi $(docker image ls -aq)
         df -h
 
-    - name: Deploy chart
+    - name: Deploy nginx chart ☕️
```
Comment (tczekajlo) [inline .github/workflows/ci.yml:95]:
```suggestion
    - name: Deploy ingress-nginx ☕️
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -88,40 +92,69 @@ jobs:
...
-        helm install \
-          --set rasax.disableTelemetry=true \
-          --set duckling.enabled=false \
-          --set rasax.initialUser.password=${INITIAL_USER_PASSWORD} \
-          --set nginx.service.type=ClusterIP \
-          --set ingress.hosts[0].host=,ingress.hosts[0].paths={/} \
-          ${RELEASE_NAME} ${CHART}
-
-    - name: Wait for deployment to be ready
+        helmfile -f .github/deployments/nginx-helmfile.yaml sync
```
Comment (tczekajlo) [inline .github/workflows/ci.yml:97]:
```suggestion
        helmfile -f .github/deployments/ingress-nginx-helmfile.yaml sync
```
You have to change the file name as well :)

**Review by tczekajlo — CHANGES_REQUESTED** [review]

**Review by tczekajlo — APPROVED** [review]
LGTM 🎖️

**Pull request merged**